react/features/authentication/middleware.ts

```diff
@@ -305,3 +279,27 @@ function _handleLogin({ dispatch, getState }: IStore) {
 
     dispatch(openTokenAuthUrl(tokenAuthServiceUrl));
 }
+
+/**
+ * Handles logout challenge. Opens logout dialog and hangs up the conference.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {string} logoutUrl - The url for logging out.
+ * @returns {void}
+ */
+function _handleLogout({ dispatch, getState }: IStore) {
+    const state = getState();
+    const { conference } = state['features/base/conference'];
+
+    const config = state['features/base/config'];
+    const logoutUrl = config.tokenLogoutUrl;
+
+    if (!conference) {
+        return;
+    }
+
+    if (logoutUrl) {
```

Comment:
This doesn't take into consideration the non token auth.

For non-token auth all we need to do is the moderator logout and hangup.